### PR TITLE
Move `notifications` package to status-go codebase.

### LINKED ISSUE #655: Remove or move notification package from go-ethereum fork to status-go
# Problem

According to some comments ([here](https://github.com/status-im/status-go/pull/648#issuecomment-364937779) and [here](https://github.com/status-im/status-go/pull/648#issuecomment-364776526)), regarding a package which deals with notifications in https://github.com/status-im/go-ethereum/tree/develop/whisper/notifications, it is not clear if this code is required, but at the very least, it is not necessary for this code to live there and it implies patches (https://github.com/status-im/status-go/blob/develop/_assets/patches/geth/0004-whisper-notifications.patch, https://github.com/status-im/status-go/blob/develop/_assets/patches/geth/0013-whisperv6-notifications.patch). Should the code still be needed, it should be moved to the status-go repo.

# Implementation

Investigate whether code is needed. Remove it if not, move it to status-go if yes.

# Acceptance Criteria

- After removal push notifications work properly.
- Chat performance and functionality is not deteriorated.

# Notes

Code that initializes notifications: https://github.com/status-im/status-go/blob/develop/geth/node/node.go#L186-L192

### PR DESCRIPTION
**Pull request by mandrigin**
Minimum viable changes to unblock https://github.com/status-im/status-go/issues/686.
`notifications` package is moved to status-go, unnecessary parts of patches are removed.

Closes #655

### CODE REVIEW

**Comment by dshulyak** [comment]
looks good, but why not to drop this packages completely, right in this change? i think it was already confirmed that it is not used

**Comment by mandrigin** [comment]
@dshulyak was it? I missed that discussion...

**Comment by JekaMas** [comment]
I think we can drop it.

**Comment by dshulyak** [comment]
@mandrigin https://github.com/status-im/status-go/pull/648#issuecomment-364937779

**Comment by mandrigin** [comment]
@dshulyak 
```go
		// enable notification service
		if whisperConfig.EnablePushNotification {
			log.Info("Register PushNotification server")

			var notificationServer notifications.NotificationServer
```
That is exactly where this moved code is used, `NotificationServer` was never in status-go, see the same file, https://github.com/status-im/status-go/blob/develop/geth/node/node.go#L22

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-go">status-im/status-go</a></strong></div><div>status-go - The Status module that consumes go-ethereum</div></blockquote>

**Comment by dshulyak** [comment]
@mandrigin maybe i misunderstood smth, but i decided that this EnabledPushNotification is not used at all

**Comment by mandrigin** [comment]
@adambabik can you help?

**Comment by dshulyak** [comment]
if we do then we need to cover it with e2e tests asap

**Comment by adambabik** [comment]
It does not take part in sending push notifications. We can remove it and also remove the place where it is used.

File: _assets/patches/geth/0013-whisperv6-notifications-envelopeversion.patch [hunk 1]
```diff
@@ -1,36 +0,0 @@
-diff --git a/whisper/notifications/utils.go b/whisper/notifications/utils.go
```
Comment (pedropombeiro) [inline _assets/patches/geth/0013-whisperv6-notifications-envelopeversion.patch:1]: We can also remove the reference to this file from the `README.md`

**Comment by pedropombeiro** [comment]
Shouldn't the `shh.notify` command-line option be removed as well?

**Comment by mandrigin** [comment]
@PombeirP yeah, that exactly what linter told me :)

**Comment by pedropombeiro** [comment]
The same changes should be made to `_assets/patches/geth/0014-whisperv6-notifications.patch` and `0015-whisperv6-envelopes-tracing.patch`

**Review by pedropombeiro — APPROVED** [review]
LGTM once the whisperv6 patch files are addressed.

**Review by JekaMas — APPROVED** [review]
Great to see this PR

File: cmd/statusd/whispercfg.go [hunk 2]
```diff
@@ -13,22 +13,10 @@ import (
...
-	whisperConfig.EnablePushNotification = *enablePN
 	whisperConfig.EnableMailServer = *enableMailServer
 	whisperConfig.MinimumPoW = *minPow
 	whisperConfig.TTL = *ttl
 
-	if whisperConfig.EnablePushNotification && whisperConfig.IdentityFile == "" {
-		return nil, errors.New("notification server requires -identity file to be specified")
-	}
-
-	if whisperConfig.IdentityFile != "" {
```
Comment (adambabik) [inline cmd/statusd/whispercfg.go:26]: `IdentityFile` may be useful to make sure the nodeID stays the same.

Comment (adambabik) [reply cmd/statusd/whispercfg.go:26]: Scratch that. Let's leave it removed. If we hit some issues it should be implemented from scratch as this should not be a part of Whisper config if used as a nodeID.

**Pull request merged**